src/ophys_etl/modules/segmentation/processing_log.py

```diff
@@ -0,0 +1,320 @@
+import h5py
+import logging
+import datetime
+import matplotlib
+import numpy as np
+from pathlib import Path
+from typing import List
+
+from ophys_etl.modules.segmentation.utils import roi_utils
+from ophys_etl.types import ExtractROI
+from ophys_etl.modules.segmentation.seed.seeder import ImageMetricSeeder
+from ophys_etl.modules.segmentation.qc.seed import add_seeds_to_axes
+from ophys_etl.modules.segmentation.qc.detect import roi_metric_qc_plot
+from ophys_etl.modules.segmentation.qc.merge import roi_merge_plot
+
+logger = logging.getLogger(__name__)
+logging.captureWarnings(True)
+logging.basicConfig(level=logging.INFO)
+
+
+def timestamp_group(group: h5py.Group) -> None:
+    """adds a 'group_creation_time' utf-8 encoded timestamp to group
+
+    Parameters
+    ----------
+    group: h5py.Group
+        the group to timestamp
+
+    """
+    group.create_dataset(
+            "group_creation_time",
+            data=str(datetime.datetime.now()).encode("utf-8"))
+
+
+class SegmentationProcessingLog:
+    """I/O methods for segmentation quality-control (log) file
+    """
+
+    def __init__(self, path: Path):
+        self.path = path
+        self.steps_dataset_name = "processing_steps"
+
+    def append_processing_step(self,
+                               group_name: str,
+                               overwrite: bool = False) -> str:
+        with h5py.File(self.path, "a") as f:
+            steps = []
+            if self.steps_dataset_name in f:
+                if overwrite:
+                    logger.warn(
+                            f"overwriting dataset {self.steps_dataset_name}")
+                else:
+                    steps = [i.decode("utf-8")
+                             for i in f[self.steps_dataset_name][()]]
+                del f[self.steps_dataset_name]
+            # get unique group name, if necessary. e.g. multiple
+            # steps called 'filter'
+            increment = 0
+            for step in steps:
+                if step.startswith(group_name):
+                    increment += 1
```

Comment:
I would rather leave this as-is for simplicity, rather than trying to engineer a system that is smart enough to know what the root name is (i.e. what if someone wanted to name a step `my_3rd_step`) etc.